c++/src/MemoryPool.cc

```diff
@@ -57,6 +57,15 @@ namespace orc {
     resize(newSize);
   }
 
+  // This constructor is used to create a DataBuffer that does not need to be memset to 0.
+  // memset_tag is not used in the code, but is kept here to avoid run-time branching.
+  template <class T>
+  DataBuffer<T>::DataBuffer(MemoryPool& pool, uint64_t newSize, no_memset_tag_t)
```

Comment:
Is it much cleaner if we directly change the original constructor to use this implementation without memset? For places where memset is required, we explicitly call it to initialize the values? Thus we do not bother to add a new constructor.

Reply:
The new commit changes to this way. It can pass test. But I did not check the logic of every place where DataBuffer is constructed to ensure they do not implicitly assume the buffer is set to zero. The only place I found so far is `MapColumnWriter::add`